src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,134 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.ErrorString;
+import duke.gui.Ui;
+import duke.storage.Storage;
+import duke.task.DeadLine;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+/**
+ * AddCommand is a command that handles add instructions.
+ *
+ * @author yl-ang
+ */
+public class AddCommand extends Command {
+
+    private boolean isExit = false;
+    private Task task;
+
+    /**
+     * Constructs AddCommand object with the supplied user input and taskCommand.
+     *
+     * @param data Full user input.
+     * @param taskCommand Task command specified by the user.
+     * @throws DukeException If no task description or no date and time is supplied.
+     */
+    public AddCommand(String data, String taskCommand) throws DukeException {
+        Task task;
+
+        String[] temp = data.split(" ");
+
+        if (taskCommand.equals("todo")) {
+
+            if (temp.length == 1) {
+                throw new DukeException(ErrorString.ERROR_EMPTY_TODO_DESC.toString());
+            }
+
+            String desc = extractDesc(data, "todo ", null);
+            task = new ToDo(desc);
+
+        } else if (taskCommand.equals("deadline")) {
+
+            if ((temp.length == 1) || (temp[1].equals("/by"))) {
+                throw new DukeException(ErrorString.ERROR_EMPTY_DEADLINE_DESC.toString());
+            }
+
+            if (!data.contains("/by")) {
+                throw new DukeException(ErrorString.ERROR_EMPTY_DEADLINE_DATE.toString());
+            }
+
+            String desc = extractDesc(data, "deadline ", "/by");
+            String date = extractTime(data, "/by ");
+
+            task = new DeadLine(desc, date);
+
+        } else {
+
+            // Event handling
+
+            if ((temp.length == 1) || (temp[1].equals("/at"))) {
+                throw new DukeException(ErrorString.ERROR_EMPTY_EVENT_DESC.toString());
+            }
+
+            if (!data.contains("/at")) {
+                throw new DukeException(ErrorString.ERROR_EMPTY_EVENT_DATE.toString());
+            }
+
+            String desc = extractDesc(data, "event ", "/at");
+            String date = extractTime(data, "/at ");
+            task = new Event(desc, date);
+        }
```

Comment:
Might be better to have event handling in another else if branch since its an if condition that can be easily specified. 
Helps with less error-prone code quality.